src/genome.py

```diff
@@ -126,7 +234,13 @@ def copy_te(self, te: int, offset: int) -> int | None:
 
         If te is not active, return None (and do not copy it).
         """
-        ...  # FIXME
+        if te in self.te.keys(): # checks if te-ID is in the te dictionary
```

Comment:
You don't need `.keys()` here. For `dicts` you can check if a key is in the dictionary with `key in dict`.